File: 287-find-the-duplicate-number/leochang/287.cpp
class Solution {
public:
	int findDuplicate(vector<int>& nums) {
		int low = 1, high = nums.size() - 1;
		while (low < high) {
			int mid = (low + high) / 2, count = 0;
			for (auto num : nums) {
				if (num <= mid) {
					count++;
				}
			}
			if (count > mid) {
				high = mid;
			} else {
				low = mid + 1;
			}
		}
		return low;
	}
};

File: 287-find-the-duplicate-number/ken/287.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        start = 1
        end = len(nums) - 1

        while start < end:
            m = (end + start) / 2
            cnt = 0

            for num in nums:
                if num <= m:
                    cnt += 1

            if cnt > m:
                end = m
            else:
                start = m + 1

        return start
